repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/services/opentracing/OpenTracingServiceConstOffHttpTest.groovy

```diff
@@ -0,0 +1,215 @@
+        when: "Request is sent through repose"
+        def messageChain = deproxy.makeRequest(
+            url: reposeEndpoint,
+            headers: [(TRACING_HEADER): trace_id ],
+            method: method)
+
+        then: "The request should have reached the origin service"
+        messageChain.handlings.size() == 1
+
+        and: "request should have 2 tracer headers"
+        if (trace_id == null)
+            assert messageChain.handlings.get(0).request.headers.getCountByName(TRACING_HEADER) == 1
+        else
+            assert messageChain.handlings.get(0).request.headers.getCountByName(TRACING_HEADER) == 2
+
+        and: "request should have tracer header pass through as well as a new header added"
+        def newTraceId
+        if (trace_id == null) {
+            newTraceId = messageChain.handlings.get(0).request.headers.getFirstValue(TRACING_HEADER)
+        } else {
+            def validateCount = 0
+            messageChain.handlings.get(0).request.headers.each {
+                if (it.name == TRACING_HEADER) {
+                    if (it.value == trace_id) validateCount++
+                    else newTraceId = it.value
+                }
+            }
+            assert validateCount == 1
+        }
+
+
+        and: "Repose should return with a 200"
+        messageChain.receivedResponse.code == "200"
+
+        and: "trace id does not exist in the new trace"
+        assert newTraceId != trace_id
+
+        where:
+        method   | trace_id
+        "HEAD"   | 'fake'
+        "GET"    | 'fake'
+        "PUT"    | 'fake'
+        "POST"   | 'fake'
+        "PATCH"  | 'fake'
+        "DELETE" | 'fake'
+        "TRACE"  | 'fake'
+        "HEAD"   | null
+        "GET"    | null
+        "PUT"    | null
+        "POST"   | null
+        "PATCH"  | null
+        "DELETE" | null
+        "TRACE"  | null
+        "HEAD"   | null
+    }
+
+
+    @Unroll("When const sample=off and parent span, should return 200 with #method and #trace_id")
+    def "when OpenTracing config is enabled with const sample=off, with parent span, trace information is passed in tracing header"() {
+
+        when: "Request is sent through repose"
+        def messageChain = deproxy.makeRequest(
+            url: reposeEndpoint,
+            headers: [(TRACING_HEADER): trace_id ],
+            method: method)
+
+        then: "The request should have reached the origin service"
+        messageChain.handlings.size() == 1
+
+        and: "request should have 2 tracer headers"
+        messageChain.handlings.get(0).request.headers.getCountByName(TRACING_HEADER) == 2
+
+
+        and: "request should have tracer header pass through as well as a new header added"
+        def newTraceId
+        def validateCount = 0
+        def validateUniqueList = []
+        messageChain.handlings.get(0).request.headers.each {
+            if (it.name == TRACING_HEADER) {
+                if (!validateUniqueList.contains(it.value)) validateUniqueList << it.value
```

Comment:
not a deal breaker but `{`/`}`s would be preferable here. I'm surprised this passed codenarc.